Sources/KeyboardShortcuts/SwiftUI+Extensions.swift

```diff
@@ -0,0 +1,123 @@
+// Created by Neil Clayton on 17/07/2024.
+// Using ideas by mbenoukaiss, https://github.com/sindresorhus/KeyboardShortcuts/issues/101
+
+
+import Foundation
+import SwiftUI
+import KeyboardShortcuts
+import Carbon
+
+// Provides a SwiftUI like wrapper func, that feels the same as the normal SwiftUI keyboardShortcut
+@available(macOS 11.0, *)
+extension View {
+	@ViewBuilder
+	public func keyboardShortcut(_ shortcutName: KeyboardShortcuts.Name) -> some View {
+		KeyboardShortcutView(shortcutName: shortcutName) {
+			self
+		}
+	}
+}
+
+// Holds the state of the shortcut, and changes that state when the shortcut changes
+// This causes the related NSMenuItem to also update (yipeee)
+@available(macOS 11.0, *)
+struct KeyboardShortcutView<Content: View>: View {
```

Comment:
I think the idiomatic way to do this would be to use `ViewModifier` instead

Reply:
Yup - there's a key modifier directly above.  I'm only using KeyboardShortcutView as a wrapper to store state so that the view/menu changes when the shortcut changes.  If this is possible using a viewModifier function, happy to learn about it and adjust.

Reply:
No, that's a `View` extension, not a `ViewModifier`.